File: cyclomaticComplexity.js
// metrics/cyclomaticComplexity.js
function calculateCyclomaticComplexity(code) {
    const decisionKeywords = ['if', 'for', 'while', 'case', 'catch', '&&', '||', '?'];
    let complexity = 1; // Start from 1

    decisionKeywords.forEach(keyword => {
        const regex = new RegExp(`\\b${keyword}\\b`, 'g');
        const matches = code.match(regex);
        if (matches) {
            complexity += matches.length;
        }
    });

    return complexity;
}

module.exports = calculateCyclomaticComplexity;

File: lecture6/cyclomaticComplexity.py
# metrics/cyclomatic_complexity.py

import re

def calculate_cyclomatic_complexity(code):
    decision_keywords = ['if', 'for', 'while', 'case', 'catch', 'and', 'or', '?']
    complexity = 1  # Start from 1

    for keyword in decision_keywords:
        matches = re.findall(r'\b' + re.escape(keyword) + r'\b', code)
        complexity += len(matches)

    return complexity
